Source/control.cpp

```diff
@@ -1617,20 +1617,21 @@ void FreeControlPan()
 	MemFreeDbg(pGBoxBuff);
 }
 
-int control_WriteStringToBuffer(char *str)
+BOOL control_WriteStringToBuffer(BYTE *str)
 {
-	signed int v1;    // edx
-	unsigned char v2; // al
+	int k;
+	BYTE ichar;

```

Comment:
Was it really need to use `BYTE` for bin-exactness here?

It's quite strange to look at the callers using casts like `(BYTE *)istr`.

Reply:
It's needed to lookup the upper chars in the translit table (gbFontTransTbl), you can tell that it was unsigned by how movzx vs movsx appears int the asm. Once the caller is cleaned up it will probably also turn out to use unsigned and so not have the odd casting, but this is the best way to clean things up one function at a time.